### PR TITLE
Add fakeovirt with imageio 

### PR DESCRIPTION
**Pull request by mnecas**
Add ovirt-imageio server with:
- generated certs
- specific config to start up with the certs
- custom ticket to add after starting
- cirros image to which the tickets points

Add fakeovirt stubs for xml and json requests and custom stubs.json for posts to create imagetransfer.
For the stubs we need to use follow params like: `vms?follow=diskattachments,nics,host_devices,snapshots,watchdogs,cdroms` `networks?follow=vnic_profiles` `hosts?follow=network_attachments,nics`

### CODE REVIEW

**Comment by ahadas** [comment]
@aesteve-rh

File: ovirt/fakeovirt_deployment.yml
Comment (aesteve-rh) [inline ovirt/fakeovirt_deployment.yml:36]: This replicates what you can achieve with the containerized imageio server that is provided at https://quay.io/repository/ovirt/ovirt-imageio.

This container serves an image, automatically generating and using a ticket, at the http port 80 (non-secure connection).

```
podman run \                           
        --interactive \
        --tty \
        --rm \
        quay.io/ovirt/ovirt-imageio:latest --help
usage: entrypoint.py [-h] [--ticket-id TICKET_ID] image

ovirt-imageio

positional arguments:
  image                 Path to image.

options:
  -h, --help            show this help message and exit
  --ticket-id TICKET_ID
                        Optional. Set the ID of the ticket. It is recommended to use with random string for better security. Defaults to the image filename.
```

It mostly only takes the path (in the container) to the image that we want to serve, and allows specifying a ticket id.

So it could be something like (untested, just an approximate example):
```
      - name: imageio
        image: quay.io:ovirt/ovirt-imageio:latest
        ports:
        - containerPort: 12345:80
        volumes:
        - ovirt-imageio/cirros.raw:/images/cirros.raw
        command: ["/images/cirros.raw", "--ticket-id", "cirros"]
```
To be more clear, this is how it would look like in the command line:
```
$ podman run \                           
        --interactive \
        --tty \
        --rm \
        --publish 12345:80 \
        --volume ovirt-imageio/:/images/:Z \
        quay.io/ovirt/ovirt-imageio:latest /images/cirros.raw --ticket-id cirros
```
Which serves the image cirros.raw at `http://localhost:12345/images/cirros`.
At the moment, we only support non-secure connections. We are also adding `--insecure` option to `ovirt-img` to nicely support this option from both server and client side.

Or maybe another option would be to build another image FROM the ovirt-imageio container, and mount and run the image inside it, instead of using it directly here.

In any case, it would be great if you could try to integrate this containerized server, rather than replicating efforts. If there is any proposal for improvement or need for more configuration options, we can work those out and improve the container.

Comment (mnecas) [reply ovirt/fakeovirt_deployment.yml]: Thanks a lot for the examples!  I have chosen to build the image with the cirros.
Can you please check out it out again if you have some comments?

File: ovirt-imageio/Dockerfile
Comment (aesteve-rh) [inline ovirt-imageio/Dockerfile]: The `ovirt-imageio` image has an `ENTRYPOINT`, which can be combined with `CMD` to pass arguments.

So something like this should work I think:
```
CMD [ "/images/cirros.raw", "--ticket-id", "cirros" ]
```

Writing a command when running this image would overwrite the `CMD` options, but not the `ENTRYPOINT` launcher.

File: ovirt/fakeovirt_deployment.yml
Comment (aesteve-rh) [inline ovirt/fakeovirt_deployment.yml:31]: The `"/app/entrypoint.py"` part should not be necessary, that's the container entrypoint, is never overridden by the command.

Actually, having a `CMD` in the Dockerfile would allow to completely skip the `command` here, if you opt for that option.

File: ovirt/stubs/imagetransfers/stub.json
Comment (aesteve-rh) [inline ovirt/stubs/imagetransfers/stub.json:27]: Should this be port 80? If you prefer to keep 12345, you can expose that port in the `Dockerfile` and the `fakeovirt_deployment.yml`, but whatever you pick, it needs to be consistent.

**Review by aesteve-rh — CHANGES_REQUESTED** [review]
It looks better! I'm glad to see that the containerized imageio server works, and saves many files and efforts :)

Few tweaks and suggestions. Also, unless you'd like to keep the files in the git history, I would squash the first and last commits.

File: ovirt/stubs/imagetransfers/stub.json
Comment (mnecas) [reply ovirt/stubs/imagetransfers/stub.json:27]: What I understand the imageio server serves the images on port 80 so we need to map it to the 12345

File: ovirt/fakeovirt_deployment.yml
Comment (mnecas) [reply ovirt/fakeovirt_deployment.yml:31]: yeah good point, I was focusing too much on this command instead of `CMD` in the image.
And in the `command` the prefix `/app/entrypoint.py"` does not work but in the `CMD` works

File: ovirt/stubs/imagetransfers/stub.json
Comment (aesteve-rh) [reply ovirt/stubs/imagetransfers/stub.json:27]: Maybe I misunderstood the line. The containerized server exposes port 80 to serve the image, yes. So where the `12345` comes from in this? This is not trying to communicate with the imageio server?

Comment (mnecas) [reply ovirt/stubs/imagetransfers/stub.json:27]: It's from the service
```
  - name: imageio
    port: 12345
    targetPort: 80
```

**Comment by aesteve-rh** [comment]
btw, we have already published some documentation regarding the containerized imageio: https://github.com/oVirt/ovirt-imageio/blob/master/docs/container.md

More or less what I did already write, but more detailed. Any updates we do will get documented there. Also, again, any request you may have to make it easier to use, or enhancement, you can either reach me directly or post an Issue in the repository.

**Review by ahadas — COMMENTED** [review]
there was a 60 sec sleep before that I wrote a comment on and seems to disappear, right? :)

File: ovirt/forklift_provider.yml
Comment (mnecas) [reply ovirt/forklift_provider.yml:14]: In the fakeovirt https://github.com/mnecas/fakeovirt

File: ovirt/patch.yml
Comment (mnecas) [reply ovirt/patch.yml:12]: yup we can drop it

**Comment by mnecas** [comment]
> there was a 60 sec sleep before that I wrote a comment on and seems to disappear, right? :) Edit: ok, I saw that comment is 'outdated'

The sleep there was just for testing purposes to see if the migration passed.

File: ovirt/forklift_provider.yml
Comment (mnecas) [reply ovirt/forklift_provider.yml:14]: sure so it's more "official"
I have create fork from my fork which has all the necessary code 
https://github.com/kubev2v/fakeovirt

Comment (mnecas) [reply ovirt/forklift_provider.yml:14]: With this, we will need to have some repos in quay under kubev2v... this reminds me that we should add the rebuild and push after patch is merged

File: ovirt/fakeovirt_deployment.yml
Comment (mnecas) [reply ovirt/fakeovirt_deployment.yml]: Splitted to two deployments one I named the `fakeovirt` the other `imageio` not the most creating but I'm opened to any suggestions

**Review by liranr23 — APPROVED** [review]
LGTM, once the e2e test in go will be written we will need to remove the related yamls and switch from the script to the forklift makefile command.

File: ovirt/forklift_provider.yml
Comment (ahadas) [reply ovirt/forklift_provider.yml:14]: yeah, currently quay monitors changes in the repos and builds the images from the Dockerfiles so we need to replace that with a mechanism like you wrote

but about fakeovirt - why not to move the repo from your account to kubev2v and then fork it?

File: ovirt/fakeovirt_deployment.yml
Comment (ahadas) [reply ovirt/fakeovirt_deployment.yml]: I meant to rename the deployment that includes both imageio and fakeovirt but splitting them as you did is also ok

File: ovirt/forklift_provider.yml
Comment (mnecas) [reply ovirt/forklift_provider.yml:14]: Updated the repo.
What do you mean by "quay monitors changes"?
AFAIK the quay does not have any feature like this but we need to push to it with our CI.

Comment (ahadas) [reply ovirt/forklift_provider.yml:14]: > AFAIK the quay does not have any feature like this but we need to push to it with our CI.

yeah, I didn't know it's possible too - but apparently there's a pattern that you can set and then quay monitors branches that match that pattern and trigger building and pushing images using the Dockerfile it finds in the repo

**Pull request merged**
